packages/xo-server/src/xapi/index.js

```diff
@@ -1157,7 +1157,8 @@ export default class Xapi extends XapiBase {
         vifsMap,
         {
           force: 'true',
-        }
+        },
+        {} // VGPU map
```

Comment:
Is this a new param?
If yes, will it break old XAPI to use it?

What's the issue of not sending this param?

Reply:
An error in some cases that provides a migration fail:
```
/var/log/xensource.log.1:35182:Jul 15 15:06:55 <host> xapi: [ warn|<host>|178440 |Async.VM.migrate_send R:3793418e587b|rbac_audit] cannot marshall arguments for the action VM.migrate_send: name and value list lengths don't match. str_names=[session_id,vm,dest,live,vdi_map,vif_map,options,vgpu_map,], xml_values=[S(OpaqueRef:934aaee6-cf8e-436f-8396-ae93d8745db6),S(OpaqueRef:06934e5f-04ca-4ff9-96a8-fe8a244d4fce),{SM:S(http://<ip>/services/SM?session_id=OpaqueRef:2b31dd59-acf2-473c-8c04-c3040b8a1301);host:S(OpaqueRef:076a9aca-5df0-47dc-a050-21fbb96abfda);xenops:S(http://<ip>/services/xenops?session_id=OpaqueRef:2b31dd59-acf2-473c-8c04-c3040b8a1301);session_id:S(OpaqueRef:2b31dd59-acf2-473c-8c04-c3040b8a1301);master:S(http://<ip>/)},B(true),{OpaqueRef:b622ba46-ea56-4a27-80a4-fce6e5e6d7d3:S(OpaqueRef:b9c12444-c03d-4808-980c-d64c220465c4)},{},{force:S(true)},]
```

Reply:
The params seems to exist since XS6.1.

When not provided it sometimes (not yet understand why sometimes) show an error in the logs and the migrate task end in error:
```
Jul 15 14:41:43 xcp-ng-sam1 xapi: [ warn|xcp-ng-sam1|170311 |Async.VM.migrate_send R:f26d7031bda6|rbac_audit] cannot marshall arguments for the action VM.migrate_send: name and value list lengths don't match. str_names=[session_id,vm,dest,live,vdi_map,vif_map,options,vgpu_map,], xml_values=[S(OpaqueRef:934aaee6-cf8e-436f-8396-ae93d8745db6),S(OpaqueRef:48380b48-d753-4f0f-9f0f-0dc4a1e60acf),{SM:S(http://62.210.84.150/services/SM?session_id=OpaqueRef:741dffac-a060-4b20-8aef-f34e7f7606eb);host:S(OpaqueRef:96ad966c-7977-4764-806d-7b3a37680b5d);xenops:S(http://62.210.84.150/services/xenops?session_id=OpaqueRef:741dffac-a060-4b20-8aef-f34e7f7606eb);session_id:S(OpaqueRef:741dffac-a060-4b20-8aef-f34e7f7606eb);master:S(http://62.210.84.150/)},B(true),{OpaqueRef:60786073-1c6f-4a8a-89d8-4094a0cd301f:S(OpaqueRef:b9c34f65-c0fb-442f-ba2f-c20d580ce544)},{},{force:S(true)},]
```

Reply:
I think it's safe to drop the compat for XS < 6.1 for this, we'll work around it if some users need it.